internal/pkg/selinux/selinux.go

```diff
@@ -6,54 +6,84 @@
 package selinux
 
 import (
+	"bytes"
 	_ "embed"
 	"log"
 	"os"
 
+	"github.com/pkg/xattr"
 	"github.com/siderolabs/go-procfs/procfs"
-	"golang.org/x/sys/unix"
 
 	"github.com/siderolabs/talos/pkg/machinery/constants"
 )
 
 //go:embed policy/policy.33
 var policy []byte
 
+var isEnabled, checkedEnabled, isEnforcing, checkedEnforcing bool
```

Comment:
I know this code is sequential, but I really don't like the idea of the global mutable variables. Also `is|checked` combo can be replaced with `optional.Optional[bool]`.

Reply:
Another option could be defining functions as constants using [OnceValue](https://pkg.go.dev/sync#OnceValue). To me global variable also seems not a really great idea, so perhaps this is more atomic.

Reply:
Great job

Reply:
I think `IsEnabled`/`IsEnforcing` is better as `sync.OnceValue` - we can't guarantee they won't be called concurrently.

Reply:
I agree. I think you can store `sync.OnceValue` inside unexported variable, move logic there and call this variable from the existing function.

Reply:
Yes, will do like this